indico/modules/events/papers/client/js/components/PaperDecisionForm.jsx

```diff
@@ -0,0 +1,94 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {useCallback} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {Form as FinalForm} from 'react-final-form';
+import {Form} from 'semantic-ui-react';
+
+import {FinalDropdown, FinalSubmitButton, FinalTextArea} from 'indico/react/forms';
+import {Translate} from 'indico/react/i18n';
+
+import {judgePaper} from '../actions';
+import {
+  canJudgePaper,
+  getCurrentUser,
+  getPaperDetails,
+  isJudgingInProgress,
+  isEventLocked,
+} from '../selectors';
+import UserAvatar from './UserAvatar';
+
+import './PaperDecisionForm.module.scss';
+
+const actionOptions = [
+  {
+    value: 'accept',
+    text: Translate.string('Accept'),
+  },
+  {
+    value: 'reject',
+    text: Translate.string('Reject'),
+  },
+  {
+    value: 'to_be_corrected',
+    text: Translate.string('To be corrected'),
+  },
+];
+
+export default function PaperDecisionForm() {
+  const dispatch = useDispatch();
+  const {
+    event: {id: eventId},
+    contribution: {id: contributionId},
+  } = useSelector(getPaperDetails);
+  const isLocked = useSelector(isEventLocked);
+  const currentUser = useSelector(getCurrentUser);
+  const isJudging = useSelector(isJudgingInProgress);
+  const canJudge = useSelector(canJudgePaper);
+
+  const submitPaperJudgment = useCallback(
+    formData => {
+      dispatch(judgePaper(eventId, contributionId, formData));
+    },
+    [dispatch, eventId, contributionId]
+  );
+
+  if (isLocked || !canJudge) {
```

Comment:
Nope, I think that I should move it ;)

Reply:
Ah, actually I now remember why I put it inside `PaperDecisionForm`. This boils down to the fact that `useSelector(isEventLocked)` or `useSelector(canJudgePaper)` will fail as long as the paper details have not been loaded. So in order to make it work I would have to either fix the selector (so that it first checks whether the required data have been loaded) or put it inside `PaperDecisionForm`.